File: code_combat/7_Sarven_Desert/353-Wishing_Well/wishing_well.py
less = "Nimis";
more = "Non satis";
requiredGold = 104;

def sumCoinValues(coins):
    totalValue = 0;
    for coin in coins:
        totalValue += coin.value
    return totalValue


def collectAllCoins():
    item = hero.findNearest(hero.findItems())
    while item:
        hero.moveXY(item.pos.x, item.pos.y)
        item = hero.findNearest(hero.findItems())


while True:
    items = hero.findItems()
    goldAmount = sumCoinValues(items)
    if goldAmount != 0:
        if goldAmount < requiredGold:
            hero.say(less)
        elif goldAmount > requiredGold:
            hero.say(more)
        else:
            collectAllCoins()

File: projects/code_combat/7_Sarven_Desert/353-Wishing_Well/wishingWell.js
// You need exactly 104 gold. 

var less = "Nimis";
var more = "Non satis";
var requiredGold = 104;

// This function calculates the sum of all coin values.
function sumCoinValues(coins) {
    var coinIndex = 0;
    var totalValue = 0;
    // Iterate all coins.
    while (coinIndex < coins.length) {
        totalValue += coins[coinIndex].value;
        coinIndex++;
    }
    return totalValue;
}

function collectAllCoins() {
    var item = hero.findNearest(hero.findItems());
    while (item) {
        hero.moveXY(item.pos.x, item.pos.y);
        item = hero.findNearest(hero.findItems());
    }
}

while (true) {
    var items = hero.findItems();
    // Get the total value of coins.
    var goldAmount = sumCoinValues(items);
    // If there are coins, then goldAmount isn't zero.
    if (goldAmount !== 0) {
        // If goldAmount is less than requiredGold
        if (goldAmount < requiredGold) {
            // Then say "Non satis".
            hero.say("Non satis");
        }
        // If goldAmount is greater than requiredGold
        else if (goldAmount > requiredGold) {
            // Then say "Nimis".
            hero.say("Nimis");
        }
        // If goldAmount is exactly equal to requiredGold
        else {
            // Then collect all coins:
            collectAllCoins();
        }
    }
}
